language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/definition/DefinitionTest.java

```diff
@@ -0,0 +1,240 @@
+                                Paths.get(balPath1).toUri().toString(), balPath1, balFile1Content)},
+                {new Position(9, 15),
+                        new DefinitionTestDataModel("globalVariableInAnotherFile.json",
+                                Paths.get(balPath1).toUri().toString(), balPath2, balFile2Content)}
+        };
+    }
+
+    @DataProvider(name = "localVariablePositions")
+    public Object[][] getLocalVariablePositions() {
+        return new Object[][]{
+                {new Position(45, 9),
+                        new DefinitionTestDataModel("localVariableInFunction.json",
+                                Paths.get(balPath1).toUri().toString(), balPath1, balFile1Content)},
+                {new Position(49, 12),
+                        new DefinitionTestDataModel("localVariableInIfStatement.json",
+                                Paths.get(balPath1).toUri().toString(), balPath1, balFile1Content)},
+                {new Position(38, 10),
+                        new DefinitionTestDataModel("localVariableInForeachStatement.json",
+                                Paths.get(balPath1).toUri().toString(), balPath1, balFile1Content)},
+                {new Position(37, 25),
+                        new DefinitionTestDataModel("localVariableOnForeachStatement.json",
+                                Paths.get(balPath1).toUri().toString(), balPath1, balFile1Content)},
+                {new Position(37, 25),
+                        new DefinitionTestDataModel("localVariableOfStruct.json",
+                                Paths.get(balPath1).toUri().toString(), balPath1, balFile1Content)}
+        };
+    }
+
+    @AfterClass
+    public void cleanSamples() throws IOException {
+        org.apache.commons.io.FileUtils.deleteDirectory(new File(ROOT_DIR + "samples"));
+    }
+
+    /**
+     * Get the definition response message as a string.
+     *
+     * @param position hovering position to get the definition.
+     * @return json string value of the response
+     */
+    private String getDefinitionResponseMessageAsString(Position position, String file, String fileContent)
+            throws InterruptedException {
+        TextDocumentPositionParams positionParams = new TextDocumentPositionParams();
+        TextDocumentIdentifier identifier = new TextDocumentIdentifier();
+        identifier.setUri(Paths.get(file).toUri().toString());
+        positionParams.setTextDocument(identifier);
+        positionParams.setPosition(position);
+
+        DidOpenTextDocumentParams documentParams = new DidOpenTextDocumentParams();
+        TextDocumentItem textDocumentItem = new TextDocumentItem();
+        textDocumentItem.setUri(identifier.getUri());
+        textDocumentItem.setText(fileContent);
+        documentParams.setTextDocument(textDocumentItem);
+
+        serviceEndpoint.notify("textDocument/didOpen", documentParams);
+        sleep(200);
```

Comment:
Instead of using the thread sleep, what if we rely on the handler implementation of the client?

Reply:
i will look in to it.